### PR TITLE
solution fetch movies

### PR DESCRIPTION
**Pull request by vladyslav-harkusha**
[DEMO LINK](https://vladyslav-harkusha.github.io/react_movies-list-fetch-movies/)

### CODE REVIEW

**Review by Anna-Borodaienko — APPROVED** [review]

File: src/api.ts [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-    .catch(() => ({
-      Response: 'False',
-      Error: 'unexpected error',
-    }));
-}
+const API_URL = 'https://www.omdbapi.com/';
+
+export const getMovie = async (query: string):
+Promise<MovieData | ResponseError> => {
+  const apiKey = 'd049b4a';
```
Comment (Anna-Tkachenko) [inline src/api.ts:8]: will be better to move constant outside of the function

File: src/components/FindMovie/FindMovie.tsx [hunk 2]
```diff
@@ -1,7 +1,73 @@
...
+    }
+  };
+
+  const addMovieToListHandler = () => {
+    if (movies.find(currentMovie => currentMovie.imdbId === movie?.imdbId)) {
+      setIsMovieInList(true);
+      setTitleQuery('');
+      setMovie(null);
+    } else if (movie) {
+      addMovie([...movies, movie]);
```
Comment (Anna-Tkachenko) [inline src/components/FindMovie/FindMovie.tsx:53]: function addMovie should accept only one movie, not the array with all movies

File: src/App.tsx [hunk 3]
```diff
@@ -14,7 +15,7 @@ export const App = () => {
       </div>
 
       <div className="sidebar">
-        <FindMovie />
+        <FindMovie movies={movies} addMovie={setMovies} />
```
Comment (Anna-Tkachenko) [inline src/App.tsx:18]: create separate handler for addMovie function, it shouldn't be setMovies one

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

**Review by Anna-Tkachenko — APPROVED** [review]
Good job! 🔥

**Pull request closed without merging**